django/db/models/constraints.py

```diff
@@ -599,6 +599,8 @@ def deconstruct(self):
         return path, self.expressions, kwargs
 
     def validate(self, model, instance, exclude=None, using=DEFAULT_DB_ALIAS):
+        from django.contrib.postgres.indexes import OpClass
```

Comment:
We may not be able to import `OpClass` here :thinking: I'd move it to the top with catching `ImportError`, e.g.
```python
try:
    from django.contrib.postgres.indexes import OpClass
except ImportError:
    OpClass = None
```

Reply:
Since `OpClass` is in the main `django` package, shouldn't it always be importable, even if `django.contrib.postgres` is not in `INSTALLED_APPS`? The fact that non-PostgreSQL tests are not broken seem to imply that it's fine, even though the application is not installed.

As for why the import is done in the method, that's because doing it at the module level caused a circular import (I should have added a comment in the PR about that, my bad).

We can still apply the logic you suggest at the method level, if the answer to my question still leads towards this suggestion.

Reply:
It can be broken in the future if `django.contrib.postgres.indexes` will import/use anything that requires `psycopg`.

Reply:
That's why we have some [workarounds](https://github.com/django/django/blob/main/tests/postgres_tests/fields.py) in tests.